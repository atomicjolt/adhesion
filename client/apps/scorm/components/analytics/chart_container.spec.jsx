import React from 'react';
import { shallow } from 'enzyme';
import ChartContainer from './chart_container';

describe('Scorm Analytics ChartContainer', () => {

  let result;
  let props;
  describe('passed charts', () => {
    beforeEach(() => {
      props = {
        selected: 'passed',
        data: {
          passFail: [],
        },
      };

      result = shallow(<ChartContainer {...props} />);
    });

    it('matches the snapshot', () => {
      expect(result).toMatchSnapshot();
    });

    it('renders a single chart', () => {
      expect(result.find('Chart').length).toBe(1);
    });
  });

  describe('completed charts', () => {
    beforeEach(() => {
      props = {
        selected: 'complete',
        data: {
          completed: [],
        },
      };

      result = shallow(<ChartContainer {...props} />);
    });

    it('renders a single chart', () => {
      expect(result.find('Chart').length).toBe(1);
    });
  });
});
