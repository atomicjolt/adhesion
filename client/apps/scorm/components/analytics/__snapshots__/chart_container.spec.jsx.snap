// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scorm Analytics ChartContainer passed charts matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChartContainer
    data={
      Object {
        "passFail": Array [],
      }
    }
    selected="passed"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Chart
        colors={
          Array [
            "#67a9cf",
            "#c9c9c9",
            "#ef8a62",
          ]
        }
        data={Array []}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "colors": Array [
          "#67a9cf",
          "#c9c9c9",
          "#ef8a62",
        ],
        "data": Array [],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Chart
          colors={
            Array [
              "#67a9cf",
              "#c9c9c9",
              "#ef8a62",
            ]
          }
          data={Array []}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "colors": Array [
            "#67a9cf",
            "#c9c9c9",
            "#ef8a62",
          ],
          "data": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
